packages/hoppscotch-app/components/realtime/Mqtt.vue

```diff
@@ -345,6 +608,116 @@ const getI18nError = (error: MQTTError): string => {
   }
 }
 const clearLogEntries = () => {
-  log.value = []
+  logs.value = []
+}
+
+const currentTabId = ref("all")
+const defaultTab = {
+  id: "all",
+  name: "All Topics",
+  color: "var(--accent-color)",
+  removable: false,
+}
+const tabs = ref([defaultTab])
+
+const dragOptions = computed(() => ({
+  group: "tabs",
+  animation: 250,
+  handle: ".tab",
+  draggable: ".tab",
+  ghostClass: "cursor-move",
+}))
+
+const tabsWidth = computed(() => ({
+  maxWidth: `${tabs.value.length * 184}px`,
+  width: "100%",
+  minWidth: "0px",
+  transition: "max-width 0.2s",
+}))
+
+const isActiveTab = (id: string) => id === currentTabId.value
+const changeTab = (id: string) => {
+  currentTabId.value = id
+  if (currentTabId.value !== "all") {
+    currentTabLogs.value = logs.value.filter((log) => {
+      return log.prefix?.includes(id)
+    })
+  }
+}
+
+const addTab = (subscription: MQTTSubscription) => {
+  const { topic, color } = subscription
+  if (tabs.value.some((tab) => tab.id === topic)) {
+    changeTab(topic)
+  }
+
+  tabs.value.push({
+    id: topic,
+    name: topic,
+    color,
+    removable: true,
+  })
+  changeTab(topic)
```

Comment:
This is going to change the tab to the topic id if one of the tabs match. Okay, that sorta makes sense I think, but then there is no returns going on here, the push will happen even if something matched.